File: client/src/main/java/de/qabel/desktop/ui/tray/TrayProxy.java
package de.qabel.desktop.ui.tray;

public class TrayProxy implements QabelTray {
    private QabelTray instance;

    @Override
    public void install() {
        if (instance != null) {
            instance.install();
        }
    }

    @Override
    public void showNotification(TrayNotification notification) {
        if (instance != null) {
            instance.showNotification(notification);
        }
    }

    public void setInstance(QabelTray instance) {
        this.instance = instance;
    }
}

File: client/src/main/java/de/qabel/desktop/ui/tray/TrayProxy.kt
package de.qabel.desktop.ui.tray

class TrayProxy : QabelTray {
    var instance: QabelTray? = null

    override fun install() = instance?.install() ?: Unit

    override fun showNotification(trayNotification: TrayNotification) = instance?.showNotification(trayNotification)
        ?: throw IllegalArgumentException("notification must not be null")
}
